### PR TITLE
Blake - more cleanup

### PR DESCRIPTION
**Pull request by BlakeGuilloud**
Cleaning up more of the React app to follow modern convention.

Separating ajax requests into an Actions file, cleaning up constructor syntax, destructuring props, and various other fixes.

Also adds in nice little travis build status :D 

[Build Status](https://travis-ci.org/codeforgso/GoVote)

### CODE REVIEW

File: src/components/AboutModal.jsx [hunk 1]
```diff
@@ -7,37 +7,30 @@ import {
...
+      </Modal.Header>
+      <Modal.Body>
+        <h4>Code for Greensboro works to improve the way our community uses its data and the web.  And we need <strong>you</strong>—no coding required.</h4>
+        <br />
+        <p>We are a local 501(c)(3) organization affiliated with the Code for America Brigade program.
             Code for America brigades are volunteer civic tech communities that collaborate with local government and community partners to build open source technology websites, and tools.
             These projects provide greater access to, or augment existing government services, and otherwise help with local civic issues.
             Code for America supports Brigade chapters with resources, tools, and access to the wider civic technology movement.</p>
-          <p>We are developers, designers, data nerds, civically - minded people, and residents — in other words, we are this community.
+        <p>We are developers, designers, data nerds, civically - minded people, and residents — in other words, we are this community.
```
Comment (SchaeStewart) [inline src/components/AboutModal.jsx]: Nice

File: src/components/AboutModal.jsx [hunk 2]
```diff
@@ -7,37 +7,30 @@ import {
...
-    <div>
-      <Modal show={show} onHide={hide}>
-        <Modal.Header closeButton>
-          <Modal.Title>About Code for Greensboro</Modal.Title>
-        </Modal.Header>
-        <Modal.Body>
-          <h4>Code for Greensboro works to improve the way our community uses its data and the web.  And we need <strong>you</strong>—no coding required.</h4>
-          <br />
-          <p>We are a local 501(c)(3) organization affiliated with the Code for America Brigade program.
+const AboutModal = ({ show, hide }) => (
```
Comment (SchaeStewart) [inline src/components/AboutModal.jsx]: Should we add some rules to our eslint to prefer destructuring?

Comment (BlakeGuilloud) [reply src/components/AboutModal.jsx]: @SchaeStewart I would be for this. I always hate writing eslint rules / keeping them updated.. I ended up writing some configs for my work awhile back, might be worth looking into: https://github.com/rhinogram/eslint-config-rhinogram

File: src/components/AboutModal.jsx (hunk 1 above)
Comment (BlakeGuilloud) [reply src/components/AboutModal.jsx]: Wish I could take credit. I only changed the indentation 💃

File: src/components/AboutModal.jsx (hunk 2 above)
Comment (SchaeStewart) [reply src/components/AboutModal.jsx]: Okay cool. I'll dig into soon. I'm think we add rules for object destructuring and requring propTypes.

File: src/components/AboutModal.jsx (hunk 1 above)
Comment (SchaeStewart) [reply src/components/AboutModal.jsx]: Oh, my bad. I miss read it

**Pull request merged**